### ISSUE TITLE
Vault Provider Lacks IAM Assume Role Capability

### ISSUE THREAD

**Issue by jschwartzy** [hashicorp/terraform-provider-vault#1086]
### Terraform Version
`Terraform v0.14.11`

### Affected Resource(s)
Vault Provider Authentication

### Terraform Configuration Files
We have used an external script to generate credentials from sts for the assumed role:
```hcl

data "external" "aws_auth" {
  program = ["${path.cwd}/../scripts/aws_auth.sh"]

  query = {
    role_arn   = var.assume_role_arn
    aws_region = var.aws_region
  }
}

provider "vault" {
  address         = "https://vault-server"
  skip_tls_verify = false
  namespace       = "namespace

  auth_login {
    path   = "auth/aws/login"
    method = "aws"

    parameters = {
      sts_region            = "us-east-1"
      aws_access_key_id     = data.external.aws_auth.result.AccessKeyId
      aws_secret_access_key = data.external.aws_auth.result.SecretAccessKey
      aws_security_token    = data.external.aws_auth.result.SessionToken
    }
  }
}
```

### Panic Output
n/a

### Expected Behavior
Vault Provider should work similarly to the AWS Provider and support assume role 

### Actual Behavior
Vault Provider supports credentials from ~/.aws/credentials (but not from source_profile) or explicitly passed credentials in the parameter configuration

### Steps to Reproduce
1. `terraform apply`

### References
- https://github.com/hashicorp/vault/issues/5767

**Comment by vinay-gopalan** [comment]
Hi @jschwartzy , thanks for submitting the issue. Would you mind sharing the output for the TF config file above? Thanks!

**Comment by emmaroberts-nbs** [comment]
I'm having the same issue.

My Tf code looks like this:
```hcl
provider "vault" {
  address = "https://vault.example.com/"
  token_name = "test-local"
  namespace = "aws/test"
  auth_login {
    path = "auth/aws/login"
    method = "aws"
    namespace = "aws/test"
    parameters = {
      role = "test-role"
      sts_region="us-east-1"
    }
  }
}
data "vault_generic_secret" "test" {
  path = "secrets/test"
}
```

I have tried running the following with the following command
`AWS_PROFILE=admin terraform apply`

I am getting the following error:
![Error message](https://user-images.githubusercontent.com/74302098/132467722-2d43bab7-9c63-40b4-9d14-331ce85e98aa.png)

**Comment by vinay-gopalan** [comment]
> I'm having the same issue.
> 
> My Tf code looks like this:
> 
> ```terraform
> provider "vault" {
>   address = "https://vault.example.com/"
>   token_name = "test-local"
>   namespace = "aws/test"
>   auth_login {
>     path = "auth/aws/login"
>     method = "aws"
>     namespace = "aws/test"
>     parameters = {
>       role = "test-role"
>       sts_region="us-east-1"
>     }
>   }
> }
> data "vault_generic_secret" "test" {
>   path = "secrets/test"
> }
> ```
> 
> I have tried running the following with the following command
> `AWS_PROFILE=admin terraform apply`
> 
> I am getting the following error:
> ![Error message](https://user-images.githubusercontent.com/74302098/132467722-2d43bab7-9c63-40b4-9d14-331ce85e98aa.png)

@emmaroberts-nbs I used your TF config file and was able to assume a role and login with AWS by running `terraform apply`. The only time I saw the error you were seeing is when I didn't have my AWS environment variables set. Can you confirm that you have those either in your terminal or in the `~/.aws/credentials` file? 

```
export AWS_ACCESS_KEY_ID=...
export AWS_SECRET_ACCESS_KEY=...
export AWS_SESSION_TOKEN=...
```

**Comment by vinay-gopalan** [comment]
Hi @jschwartzy thanks for the additional info. This issue was tracked in Vault instead of the Vault Provider for TF, and the bug was found and resolved down the ladder in the AWS util package https://github.com/hashicorp/go-secure-stdlib/pull/11

**Comment by vinay-gopalan** [comment]
Closing this issue since fix was merged and should be released with Vault 1.9. Please feel free to open another issue if the problem persists, thanks!

**Comment by benashz** [comment]
We recommend using the first-class aws login support. See https://github.com/hashicorp/terraform-provider-vault/pull/1588#issuecomment-1278174354 for related info.
